### PR TITLE
Update to Flink 1.19

### PR DESCRIPTION
**Pull request by nicusX**
* Updated most of examples to Flink 1.19
* Reworked examples for consistency
* Substantially changed Python Getting Started, including extensive README

### CODE REVIEW

File: java/CustomMetrics/README.md [hunk 1]
```diff
@@ -1,27 +1,20 @@
 # Custom Metrics
 
-* Flink version: 1.18.1
+* Flink version: 1.19.0
```
Comment (elphastori) [inline java/CustomMetrics/README.md:3]: Now 1.19.1

Comment (nicusX) [reply java/CustomMetrics/README.md]: Good shout!. Fixed

**Review by hlteoh37 — APPROVED** [review]
LGTM - thanks for testing this @nicusX

File: scala/GettingStarted/build.sbt [hunk 2]
```diff
@@ -9,8 +9,8 @@ lazy val root = (project in file(".")).
     javacOptions ++= Seq("-source", "11", "-target", "11")
   )
 
-val jarName = "getting-started-scala-1.0.jar"
-val flinkVersion = "1.18.1"
+val jarName = "ScalaGettingStarted-flink_1.19.jar"
+val flinkVersion = "1.19.0"
```
Comment (hlteoh37) [inline scala/GettingStarted/build.sbt:13]: This should be 1.19.1

**Comment by addu390** [comment]
Hi @nicusX and @hlteoh37 
Curious on if there were any limitation to upgrade `Java` > `IcebergDatastreamSink` example to `Flink 1.19.1`?

**Comment by nicusX** [comment]
@addu390 there no specific limitation. Just a trivial matter of time and prioritization. We will update it

**Comment by addu390** [comment]
> @addu390 there no specific limitation. Just a trivial matter of time and prioritization. We will update it

Got it, thanks! I did try with `Flink 1.20`, worked out well without much of changes.

**Pull request merged**